textdb/textdb-dataflow/src/test/java/edu/uci/ics/textdb/dataflow/keywordmatch/PhraseMatcherTest.java

```diff
@@ -0,0 +1,268 @@
+        attributeList.add(TestConstants.LAST_NAME_ATTR);
+        attributeList.add(TestConstants.DESCRIPTION_ATTR);
+
+        //Prepare expected result list
+        List<Span> list = new ArrayList<>();
+        Span span1 = new Span("description", 25, 45, "lin and and angry", "lin clooney is Angry");
+
+        list.add(span1);
+
+        Attribute[] schemaAttributes = new Attribute[TestConstants.ATTRIBUTES_PEOPLE.length + 1];
+        for(int count = 0; count < schemaAttributes.length - 1; count++) {
+            schemaAttributes[count] = TestConstants.ATTRIBUTES_PEOPLE[count];
+        }
+        schemaAttributes[schemaAttributes.length - 1] = SchemaConstants.SPAN_LIST_ATTRIBUTE;
+
+        IField[] fields1 = { new StringField("george lin lin"), new StringField("lin clooney"), new IntegerField(43),
+                new DoubleField(6.06), new DateField(new SimpleDateFormat("MM-dd-yyyy").parse("01-13-1973")),
+                new TextField("Lin Clooney is Short and lin clooney is Angry"), new ListField<>(list) };
+
+        ITuple tuple1 = new DataTuple(new Schema(schemaAttributes), fields1);
+        List<ITuple> expectedResultList = new ArrayList<>();
+        expectedResultList.add(tuple1);
+
+        //Perform Query
+        List<ITuple> resultList = getPeopleQueryResults(query, attributeList);
+
+        //Perform Check
+        boolean contains = TestUtils.containsAllResults(expectedResultList, resultList);
+        Assert.assertTrue(contains);
+        Assert.assertEquals(1,resultList.size());
+    }
+
+    /**
+     * Verifies: Query with Stop Words match corresponding phrases in the document
+     * @throws Exception
+     */
+    @Test
+    public void testWordInMultipleFieldsQueryWithStopWords2() throws Exception {
+        //Prepare Query
+        String query = "lin clooney and angry";
+        ArrayList<Attribute> attributeList = new ArrayList<>();
+        attributeList.add(TestConstants.FIRST_NAME_ATTR);
+        attributeList.add(TestConstants.LAST_NAME_ATTR);
+        attributeList.add(TestConstants.DESCRIPTION_ATTR);
+
+        //Prepare expected result list
+        List<Span> list = new ArrayList<>();
+        Span span1 = new Span("description", 25, 45, "lin clooney and angry", "lin clooney is Angry");
+
+        list.add(span1);
+
+        Attribute[] schemaAttributes = new Attribute[TestConstants.ATTRIBUTES_PEOPLE.length + 1];
+        for(int count = 0; count < schemaAttributes.length - 1; count++) {
+            schemaAttributes[count] = TestConstants.ATTRIBUTES_PEOPLE[count];
+        }
+        schemaAttributes[schemaAttributes.length - 1] = SchemaConstants.SPAN_LIST_ATTRIBUTE;
+
+        IField[] fields1 = { new StringField("george lin lin"), new StringField("lin clooney"), new IntegerField(43),
+                new DoubleField(6.06), new DateField(new SimpleDateFormat("MM-dd-yyyy").parse("01-13-1973")),
+                new TextField("Lin Clooney is Short and lin clooney is Angry"), new ListField<>(list) };
```

Comment:
@prakul  Good test cases. If possible could you add one more test case where phrase matcher doesn't return any results. For eg: when query is "lin clooney is short and angry" - the phrase matcher doesn't return any results. The functionality will be more clear. :)